pumpkin-lib/src/engine/cp/assignments.rs

```diff
@@ -16,6 +16,9 @@ pub struct Assignments {
     pub(crate) trail: Trail<ConstraintProgrammingTrailEntry>,
     domains: KeyedVec<DomainId, IntegerDomain>,
     events: EventSink,
+
+    /// The number of values that have been pruned from the domain.
+    pruned_values: u64,
```

Comment:
is the intention of `pruned_values` to be an exact, or is it an approximation? Based on the code, I would think it is an approximation, please do clarify!